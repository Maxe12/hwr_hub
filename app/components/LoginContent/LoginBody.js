import React from 'react';
import { ImageBackground, AsyncStorage } from 'react-native';
import { Content, Card, CardItem, Item, Input, Container, Text, Button } from 'native-base';
import styles from './Styles';

const loginVals = {username: 'root', password: 'password'}

export default class LoginBody extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            username: '', 
            password: ''
        }
    }
    render() {
        return (
            <Container>
                <ImageBackground source={require('../../assets/images/HWR_Lichtenberg.jpg')}
                    imageStyle={{ resizeMode: 'stretch' }}
                    style={styles.backgroundImage}>
                    <Content style={styles.loginWindow}>
                        <Card header style={styles.cardStyle}>
                            <CardItem bordered style={styles.cardItemStyle}>
                                <Text style={styles.heading}>Hochschule f√ºr Wirtschaft und Recht Berlin</Text>
                            </CardItem>
                            <Item regular>
                                <Input placeholder='Username' 
                                onChangeText={((username)=>this.setState({username}))} 
                                value={this.state.username} />
                            </Item>
                            <Item regular>
                                <Input placeholder='Password'  
                                secureTextEntry
                                onChangeText={(password) => this.setState({password})} 
                                value={this.state.password} />
                            </Item>
                            <Button block 
                            style={{alignSelf: 'center', backgroundColor: 'red'}}
                            onPress={() => this.loginButtonHandler()}>
                                <Text>Login</Text>
                            </Button>
                        </Card>
                    </Content>
                </ImageBackground>
            </Container >
        );
    }

    loginButtonHandler = async () => {
        if(this.state.username === loginVals.username && this.state.password === loginVals.password){
            await AsyncStorage.setItem('isLoggedIn', '1');
            this.props.navigation.navigate('Main');
        } else {
            alert("Username/Password is wrong"); 
        }
    }
}