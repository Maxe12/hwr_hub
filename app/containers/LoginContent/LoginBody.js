import React from 'react';
import { ImageBackground, AsyncStorage } from 'react-native';
import { Content, Card, CardItem, Item, Input, Container, Text, Button } from 'native-base';
import styles from './Styles';
import { Asset } from 'expo-asset';
import { connect } from 'react-redux';

const backgroundImg = Asset.fromModule(require('../../../assets/images/HWR_Lichtenberg.jpg'));
const loginVals = { username: 'root', password: 'password' }

class LoginBody extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            userdata: {
                username: '',
                password: ''
            }
        }
    }
    render() {
        return (
            <Container>
                <ImageBackground source={backgroundImg}
                    imageStyle={{ resizeMode: 'stretch' }}
                    style={styles.backgroundImage}>
                    <Content style={styles.loginWindow}>
                        <Card header style={styles.cardStyle}>
                            <CardItem bordered style={styles.cardItemStyle}>
                                <Text style={styles.heading}>Hochschule f√ºr Wirtschaft und Recht Berlin</Text>
                            </CardItem>
                            <Item regular>
                                <Input placeholder='Username'
                                    onChangeText={((text) => this.setState({ userdata: {username: text, password: this.state.userdata.password} }))}
                                    value={this.state.userdata.username} />
                            </Item>
                            <Item regular>
                                <Input placeholder='Password'
                                    secureTextEntry
                                    onChangeText={(text) => this.setState({ userdata: {username: this.state.userdata.username, password: text} })}
                                    value={this.state.userdata.password} />
                            </Item>
                            <Button block
                                style={{ alignSelf: 'center', backgroundColor: 'red' }}
                                onPress={() => this.loginButtonHandler()}>
                                <Text>Login</Text>
                            </Button>
                        </Card>
                    </Content>
                </ImageBackground>
            </Container >
        );
    }

    loginButtonHandler = async () => {
        console.log(this.state.userdata.username);
        if (this.state.userdata.username == loginVals.username && this.state.userdata.password == loginVals.password) {
            this.addUsername(this.state.userdata.username);
            await AsyncStorage.setItem('isLoggedIn', '1');
            this.props.navigation.navigate('Main');
        } else {
            alert("Username/Password is wrong");
        }
    }

    addUsername = (username) => {
        this.props.dispatch({ type: 'ADD_USERNAME', username: username });
        this.setState({
            userdata: {
                username: '',
                pasword: ''
            }
        })
    }
}

export default connect()(LoginBody);