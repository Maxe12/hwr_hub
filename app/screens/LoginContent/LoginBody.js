import React from 'react';
import { ImageBackground, AsyncStorage, TouchableOpacity } from 'react-native';
import { CardItem, Item, Input, Container, Text, Button, Body } from 'native-base';
import styles from './Styles';
import { connect } from 'react-redux';
import { addEmail } from '../../actions/Index';

const loginVals = { username: 'root', password: 'password' }
const backgroundImg = require('../../../assets/images/HWR_Lichtenberg.jpg');

class LoginBody extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: ''
        }
    }
    render() {
        return (
            <Container>
                <ImageBackground source={backgroundImg}
                    imageStyle={{ resizeMode: 'stretch' }}
                    style={styles.backgroundImage}>
                    <CardItem header bordered style={styles.cardItemHeader}>
                        <Text style={styles.textHeader}>Hochschule f√ºr Wirtschaft und Recht</Text>
                    </CardItem>
                    <CardItem bordered style={styles.cardItemBody}>
                        <Body>
                            <Text>Bitte einloggen: </Text>
                            <Item regular style={styles.inputItems}>
                                <Input placeholder='Username'
                                    onChangeText={(text) => { this.setState({ username: text }) }}
                                    value={this.state.username} />
                            </Item>
                            <Item regular style={styles.inputItems}>
                                <Input
                                    placeholder='Passwort'
                                    secureTextEntry
                                    onChangeText={(text) => this.setState({ password: text })}
                                    value={this.state.password} />
                            </Item>
                            <Item style={styles.buttonItem}>
                                <Button block style={styles.buttonComponent}
                                    onPress={() => { this.loginButtonHandler() }} >
                                    <Text>Login</Text>
                                </Button>
                            </Item>
                            <Item style={styles.registerText}>
                                <Text>Noch nicht Registriert?</Text>
                                <TouchableOpacity
                                onPress={() => {this.props.navigation.navigate('Register')}}
                                ><Text style={styles.opacityStyle}>Hier Anmelden!</Text></TouchableOpacity>
                            </Item>
                        </Body>
                    </CardItem>
                </ImageBackground>
            </Container >
        );
    }

    loginButtonHandler = async () => {
        if (this.state.username == loginVals.username && this.state.password == loginVals.password) {
            this.addUsername(this.state.username);
            await AsyncStorage.setItem('isLoggedIn', '1');
            await AsyncStorage.setItem('username', this.state.username);
            this.props.navigation.navigate('Main');
        } else {
            alert("Username/Password is wrong");
        }
    }

    addUsername = (username) => {
        this.props.onLogin(username)
        this.setState({
            username: '',
            password: ''
        })
    }
}

mapDispatchToProps = dispatch => ({
    onLogin: (email) => {
        dispatch(addEmail(email));
    }
})

export default connect(null, mapDispatchToProps)(LoginBody);