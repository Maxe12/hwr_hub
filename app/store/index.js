import { createStore } from 'redux';

const rootReducer = () => {
    //@todo reducers in here 
}

export default store = createStore(rootReducer) 