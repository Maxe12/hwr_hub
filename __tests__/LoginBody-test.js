import 'react-native';
import React from 'react';
import LoginBody from '../app/screens/LoginContent/LoginBody';
import renderer from 'react-test-renderer';
import store from '../app/store/Index';

test('LoginBody snapChot-test', ()=>{
    const snap = renderer.create(
        <LoginBody  store={store} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
})