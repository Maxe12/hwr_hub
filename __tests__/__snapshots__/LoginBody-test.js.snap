// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBody snapChot-test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Asset {
          "_downloadCallbacks": Array [],
          "downloaded": false,
          "downloading": false,
          "hash": "md5",
          "height": 1,
          "localUri": null,
          "name": "name",
          "type": "type",
          "uri": "uri",
          "width": 1,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "resizeMode": "stretch",
          },
        ]
      }
    />
    <View
      bordered={true}
      header={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "rgba(215, 215, 212, 0.8)",
            "marginTop": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "black",
              "textAlign": "center",
            },
          ]
        }
        uppercase={false}
      >
        Hochschule f√ºr Wirtschaft und Recht
      </Text>
    </View>
    <View
      bordered={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "rgba(215, 215, 212, 0.8)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          Bitte einloggen: 
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRadius": undefined,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 2,
              "marginTop": 20,
              "opacity": 1,
              "width": 280,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Username"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 8,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRadius": undefined,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 2,
              "marginTop": 20,
              "opacity": 1,
              "width": 280,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Passwort"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 8,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 2,
              "marginTop": 30,
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#D50C2F",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
                "width": 280,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                }
              }
              uppercase={false}
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
